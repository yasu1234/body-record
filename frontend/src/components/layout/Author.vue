<script setup>
import { ref, onMounted, watch } from "vue";

const props = defineProps(["author"]);

const author = ref(null);

onMounted(() => {
  author.value = props.author;
});

watch(props, () => {
  author.value = props.author;
});
</script>

<template>
  <div class="author">
    <div
      class="author-thumbnail-image"
      v-if="author !== null && author.image_url !== null"
    >
      <img
        :src="author.image_url.url"
        alt="ユーザー"
        class="author-thumbnail-image"
      />
    </div>
    <p class="author-name" v-if="author !== null">{{ author.name }}</p>
  </div>
</template>

<style scoped>
.author {
  display: flex;
  align-items: center;
  margin-top: 10px;
  margin-left: 20px;
}
.author-thumbnail-image {
  width: 40px;
  height: 40px;
  object-fit: cover;
  border-radius: 50%;
}
.author-name {
  margin-left: 10px;
}
</style>
