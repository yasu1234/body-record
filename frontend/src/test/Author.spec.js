import { shallowMount } from "@vue/test-utils";
import Author from "@/components/layout/Author.vue";

jest.mock("axios");
jest.mock("js-cookie");
jest.mock("vue-router", () => {
  const router = {
    push: jest.fn(),
    pathname: "/",
    route: "",
    asPath: "",
    query: {},
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
  };
  return {
    useRouter: jest.fn().mockReturnValue(router),
  };
});

describe("Author", () => {
  let wrapper;
  let mockedAxiosInstance;

  const mockAuthor = {
    id: 1,
    name: "Test Author",
    image_url: {
      url: "https://example.com/author.jpg",
    },
    is_support: false,
  };

  const mockUser = {
    id: 1,
    supporter_count: 10,
  };

  beforeEach(() => {    
    wrapper = shallowMount(Author, {
      global: {
        stubs: ["router-link", "router-view"],
      },
      props: {
        author: mockAuthor,
        support: mockUser
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("SnapShotテスト(サポート済み)", () => {
    const mockAuthor = {
      id: 1,
      name: "Test Author",
      image_url: {
        url: "https://example.com/author.jpg",
      },
      is_support: true,
    };

    const mockUser = {
      id: 1,
      supporter_count: 10,
    };

    wrapper = shallowMount(Author, {
      global: {
        stubs: ["router-link", "router-view"],
      },
      props: {
        author: mockAuthor,
        support: mockUser
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it("SnapShotテスト(未サポート)", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("サポート件数が表示されている", () => {
    const supportCount = wrapper.findAll("p").at(1);
    expect(supportCount.text()).toBe("10");
    expect(wrapper.find(".author-thumbnail-image img").exists()).toBe(true);
  });
});
