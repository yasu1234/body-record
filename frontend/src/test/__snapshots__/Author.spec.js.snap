// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author SnapShotテスト(サポート済み) 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="author"
  >
    <!--v-if-->
    <!--v-if-->
  </div>
  <div
    class="support-container"
  >
    <button
      class="support-button"
    >
      <img
        alt="ユーザー"
        class="support-image"
        src=""
      />
    </button>
    <!--v-if-->
  </div>
  
</div>
`;

exports[`Author SnapShotテスト(未サポート) 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="author"
  >
    <div
      class="author-thumbnail-image"
    >
      <img
        alt="ユーザー"
        class="author-thumbnail-image"
        src="https://example.com/author.jpg"
      />
    </div>
    <p
      class="author-name"
    >
      Test Author
    </p>
  </div>
  <div
    class="support-container"
  >
    <button
      class="support-button"
    >
      <img
        alt="ユーザー"
        class="support-image"
        src=""
      />
    </button>
    <p
      class="author-name"
    >
      10
    </p>
  </div>
  
</div>
`;
